web/src/LogPane.tsx

```diff
@@ -38,6 +46,62 @@ class LogPane extends Component<LogPaneProps, LogPaneState> {
 
     window.addEventListener("scroll", this.refreshAutoScroll, { passive: true })
     window.addEventListener("wheel", this.handleWheel, { passive: true })
+    if (!this.props.isSnapshot) {
+      document.addEventListener("selectionchange", this.handleSelectionChange, {
+        passive: true,
+      })
+    }
+  }
+
+  private handleSelectionChange() {
+    let selection = document.getSelection()
+    if (selection && selection.focusNode && selection.anchorNode) {
+      let node = ReactDOM.findDOMNode(this)
+      let beginning = selection.focusNode
+      let end = selection.anchorNode
+
+      // if end is before beginning
+      if (
+        beginning.compareDocumentPosition(end) &
+        Node.DOCUMENT_POSITION_PRECEDING
+      ) {
+        // swap beginning and end
+        ;[beginning, end] = [end, beginning]
+      }
+
+      if (selection.isCollapsed) {
+        this.props.handleClearHiglight()
+      } else if (
+        node &&
+        node.contains(beginning) &&
+        node.contains(end) &&
+        !node.isEqualNode(beginning) &&
+        !node.isEqualNode(end)
+      ) {
+        let beginningLogLine = this.findLogLineID(beginning.parentElement)
+        let endingLogLine = this.findLogLineID(end.parentElement)
+
+        if (beginningLogLine && endingLogLine) {
+          this.props.handleSetHighlight({
+            beginningLogID: beginningLogLine,
+            endingLogID: endingLogLine,
+            highlightedText: selection.toString(),
+          })
+        }
+      }
+    }
+  }
+
+  findLogLineID(el: HTMLElement | null): string | null {
+    if (el === null) {
+      return null
+    } else if (el && el.className.startsWith("logLine")) {
```

Comment:
generally more robust to use the classList api
https://developer.mozilla.org/en-US/docs/Web/API/Element/classList

tho in this case it'd be even better to use a data attribute